js_v9/Block.js

```diff
@@ -78,21 +78,17 @@ function Block(fallingLane, color, iter, distFromHex, settled) {
 		if(attached === undefined)
 			attached = false;
 
-		if(this.angle > this.targetAngle) {
-			this.angularVelocity -= angularVelocityConst * MainHex.dt;
-		}
-		else if(this.angle < this.targetAngle) {
-			this.angularVelocity += angularVelocityConst * MainHex.dt;
+		var diff, sign;
+		if (sign = Math.sign(diff = this.angle - this.targetAngle)) {
+			var change = this.angularVelocity - sign * angularVelocityConst * MainHex.dt;
+			if(sign * (diff + change) > Math.abs(change))
```

Comment:
Using ``sign * (diff + change)`` instead of ``Math.abs(diff + change)`` ensures that the block has not already crossed the threshold of ``this.targetAngle``.